pandas/core/window/rolling.py

```diff
@@ -445,15 +450,18 @@ def _get_roll_func(self, func_name: str) -> Callable:
             )
         return window_func
 
-    def _get_cython_func_type(self, func: str) -> Callable:
+    def _get_cython_func_type(self, func: str) -> Tuple[Callable, str]:
         """
         Return a variable or fixed cython function type.
 
         Variable algorithms do not use window while fixed do.
         """
         if self.is_freq_type or isinstance(self.window, BaseIndexer):
-            return self._get_roll_func(f"{func}_variable")
-        return partial(self._get_roll_func(f"{func}_fixed"), win=self._get_window())
+            return self._get_roll_func(f"{func}_variable"), "variable"
```

Comment:
you don't need to return this 'variable' or 'fixed'; use `self.is_freq_type` (maybe not the best name)